mailet/standard/src/main/java/org/apache/james/transport/mailets/LogMessage.java

```diff
@@ -100,25 +187,39 @@ public void service(Mail mail) {
         }
     }
 
+    private void logAttribute(List<String> specificAttributes) {
+        if(!specificAttributes.isEmpty()){
+            logBuffer.append('\n');
+            for(String attribute : specificAttributes){
+                logBuffer.append(attribute + '\n');
+            }
+        }
+    }
+
     private void logComment() {
         if (comment != null) {
-            logger.info(comment);
+            logBuffer.append(comment);
         }
     }
 
     private void logHeaders(MimeMessage message) throws MessagingException {
         if (headers && logger.isInfoEnabled()) {
-            logger.info("\n");
+            logBuffer.append('\n');
             for (String header : Collections.list(message.getAllHeaderLines())) {
-                logger.info(header + "\n");
+                logBuffer.append(header + '\n');
+            }
+            if(!specificHeaders.isEmpty()){
+                for(String header : specificHeaders){
```

Comment:
Don't we need to read mail.perRecipientHeaders here rather than only listing configured header names?